napari/layers/surface/surface.py

```diff
@@ -403,6 +453,33 @@ def normals(self, normals: Union[dict, SurfaceNormals, None]):
             self._normals.update(normals)
         self.events.normals(value=self._normals)
 
+    @property
+    def texture(self) -> Optional[np.ndarray]:
+        return self._texture
+
+    @texture.setter
+    def texture(self, texture: np.ndarray):
+        self._texture = texture
+        self.events.texture(value=self._texture)
+
+    @property
+    def texcoords(self) -> Optional[np.ndarray]:
+        return self._texcoords
+
+    @texcoords.setter
+    def texcoords(self, texcoords: np.ndarray):
+        self._texcoords = texcoords
+        self.events.texcoords(value=self._texcoords)
```

Comment:
Maybe we should check here that they are at least arrays, otherwise it will fail on the vispy side and we end up with inconsistent state.

Reply:
I added guards here - let me know if this is what you had in mind!